src/Components/Components/src/Routing/IHostEnvironmentNavigationManager.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.Routing
+{
+    /// <summary>
+    /// An optional interface for <see cref="NavigationManager" /> implementations that must be initialized
+    /// by the host.
+    /// </summary>
+    public interface IHostEnvironmentNavigationManager
+    {
+        /// <summary>
+        /// Initializes the <see cref="NavigationManager" />.
+        /// </summary>
+        /// <param name="baseUri">The base URI.</param>
+        /// <param name="uri">The absolute URI.</param>
+        void Initialize(string baseUri, string uri);
```

Comment:
You may have considered this already, but is it possible for all the server-side NavigationManager subclasses to take these two as constructor params? It seems like they should be known up front at the point where the instance is constructed for DI.

If that was the case, would that make it possible to eliminate this interface?

Reply:
IHttpContextAccessor is the only way to do that through DI, which we want to avoid

Reply:
What Javier said.